File: georocket-server-api/src/main/java/io/georocket/index/xml/MetaIndexer.java
package io.georocket.index.xml;

import io.georocket.index.Indexer;
import io.georocket.storage.ChunkMeta;
import io.georocket.storage.IndexMeta;

/**
 * Indexes chunks based on their metadata
 * @since 1.0.0
 * @author Michel Kraemer
 */
public interface MetaIndexer extends Indexer {
  /**
   * Will be called when a chunk is being indexed
   * @param path the chunk's path in the GeoRocket store
   * @param chunkMeta the chunk metadata
   * @param indexMeta the metadata used for indexing
   */
  void onIndexChunk(String path, ChunkMeta chunkMeta, IndexMeta indexMeta);
}

File: georocket-server-api/src/main/kotlin/io/georocket/index/MetaIndexer.kt
package io.georocket.index

import io.georocket.storage.ChunkMeta
import io.georocket.storage.IndexMeta

/**
 * Indexes metadata of chunks
 * @author Michel Kraemer
 */
interface MetaIndexer {
  /**
   * Will be called when a chunk with the given [path], [chunkMeta] and
   * [indexMeta] is indexed
   */
  fun onChunk(path: String, chunkMeta: ChunkMeta, indexMeta: IndexMeta): Map<String, Any>
}
